### PR TITLE
Add Track1 code validation for MGMT pipeline

### PR DESCRIPTION
**Pull request by HarveyLink**

### CODE REVIEW

File: eng/scripts/RPMapping.json [hunk 1]
```diff
@@ -0,0 +1,503 @@
+{
```
Comment (weshaggard) [inline eng/scripts/RPMapping.json:1]: Is this file necessary? Can we eliminate or default most of it? Having a separate file to maintain almost always causes issues and a higher cost to maintain.

Comment (HarveyLink) [reply eng/scripts/RPMapping.json]: Removed the RPMapping file and do the RP folder name match by directly search readme in swagger repo.

File: eng/pipelines/templates/jobs/archetype-sdk-mgmt.yml [hunk 2]
```diff
@@ -5,6 +5,12 @@ jobs:
     steps:
       #- script: "echo $(system.pullrequest.pullrequestnumber), https://github.com/$(build.repository.id), https://github.com/$(build.repository.ID)"
       - template: /eng/pipelines/templates/steps/install-dotnet.yml
+      - script: |
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-sdk-mgmt.yml:8]:
```suggestion
      - pwsh: |
```

You can directly use a pwsh task instead of calling pwsh manually

File: eng/scripts/Track1CodeCheck.ps1 [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+    $rpIndex | ForEach-Object {
+        $path = "eng/mgmt/mgmtmetadata/$_" + "_resource-manager.txt"
+        if ($mataPath -notcontains $path) {
+            $mataPath += $path
+        }
+    }
+
+    # Install AutoRest    
+    Invoke-Block {
+        & npm install -g autorest
```
Comment (mitchdenny) [inline eng/scripts/Track1CodeCheck.ps1:130]: Perhaps a naïve question. Should we be pinning to a specific version of AutoRest here? If the generated files change as a result of the version of AutoRest we use then we would block. That might not be such a problem for master, but if we need to hotfix a prior release for any reason that could complicate the release process.

I'm not sure if this is a concern that we need to deal with though.

Comment (HarveyLink) [reply eng/scripts/Track1CodeCheck.ps1:118]: Thanks for the comment. This is indeed a potential issue which needs to be handled. We'll record and keep tracking it. 
For now, I used autorest version v2 and csharp version 2.3.90, should be stable for a while. 
If there is a special verison used in release PR, pushed to *-preview brach triggers this pipeline, causes CI failures, which are expected and I think we can manually verify and merge it.
In furture, we may setup a configuration file for the pipeline or other approaches to handle it.

Comment (mitchdenny) [reply eng/scripts/Track1CodeCheck.ps1:118]: Did you have a link to the issue?

Comment (HarveyLink) [reply eng/scripts/Track1CodeCheck.ps1:118]: https://github.com/Azure/autorest.csharp/pull/106
If we use autorest by default parameters not a specific version in pipeline, it will use csharp version 2.3.79, not the latest one. Doesn't contains changes I've mentioned above. So we set version to 2.3.90 to avoid that.

File: eng/pipelines/templates/jobs/archetype-sdk-mgmt.yml [hunk 4]
```diff
@@ -5,9 +5,11 @@ jobs:
     steps:
       #- script: "echo $(system.pullrequest.pullrequestnumber), https://github.com/$(build.repository.id), https://github.com/$(build.repository.ID)"
       - template: /eng/pipelines/templates/steps/install-dotnet.yml
-      - script: |
-          pwsh eng/scripts/Track1CodeCheck.ps1
+      - pwsh: |
+          eng/scripts/Track1CodeCheck.ps1
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-sdk-mgmt.yml:9]: Why not move the entire Track1CodeCheck to a later step instead of wrapping it like this? Does the order in which it is called have impact on the other steps?

File: eng/pipelines/templates/jobs/archetype-sdk-mgmt.yml [hunk 5]
```diff
@@ -1,17 +1,21 @@
 jobs:
   - job: Build
+    timeoutInMinutes: 80
     pool:
       vmImage: windows-2019
     steps:
       #- script: "echo $(system.pullrequest.pullrequestnumber), https://github.com/$(build.repository.id), https://github.com/$(build.repository.ID)"
-      - template: /eng/pipelines/templates/steps/install-dotnet.yml
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-sdk-mgmt.yml:7]: I think it would be a good idea to keep this template install.

File: eng/pipelines/templates/jobs/archetype-sdk-mgmt.yml [hunk 6]
```diff
@@ -46,12 +50,12 @@ jobs:
     steps:
       - template: /eng/common/pipelines/templates/steps/verify-agent-os.yml
       - script: "echo $(system.pullrequest.pullrequestnumber), http://github.com/$(build.repository.id), http://github.com/$(build.repository.ID)"
-      - template: /eng/pipelines/templates/steps/install-dotnet.yml
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-sdk-mgmt.yml:49]: I would keep this template.

File: eng/pipelines/templates/jobs/archetype-sdk-mgmt.yml (hunk 4 above)
Comment (HarveyLink) [reply eng/pipelines/templates/jobs/archetype-sdk-mgmt.yml]: Sure, the only step which could wrap this part is "Build & Package", so i put it behind current scripts. No imoact found.

**Review by allenjzhang — APPROVED** [review]
